includes/class-smaily-for-wp-widget.php

```diff
@@ -54,23 +63,28 @@ public function widget( $args, $instance ) {
 		}
 
 		// Load configuration data.
-		$table_name                 = esc_sql( $wpdb->prefix . 'smaily_config' );
-		$config                     = (array) $wpdb->get_row( "SELECT * FROM `$table_name` LIMIT 1" );
+		$api_credentials = $this->options->get_api_credentials();
+		$form_options    = $this->options->get_form_options();
+		// Data to be assigned to template.
+		$config                     = array();
+		$config['domain']           = $api_credentials['subdomain'];
+		$config['form']             = $form_options['form'];
+		$config['is_advanced']      = $form_options['is_advanced'];
 		$config['show_name']        = $show_name;
 		$config['success_url']      = $success_url;
 		$config['failure_url']      = $failure_url;
 		$config['autoresponder_id'] = $autoresponder;
 
 		// Create admin template.
-		$file     = ( isset( $config['is_advanced'] ) && '1' === $config['is_advanced'] ) ? 'advanced.php' : 'basic.php';
+		$file     = $config['is_advanced'] === '1' ? 'advanced.php' : 'basic.php';
 		$template = new Smaily_For_WP_Template( 'public/partials/smaily-for-wp-public-' . $file );
 		$template->assign( $config );
 		// Display responses on Smaily subscription form.
 		$form_has_response  = false;
 		$form_is_successful = false;
 		$response_message   = null;
 
-		if ( ! isset( $config['api_credentials'] ) || empty( $config['api_credentials'] ) ) {
+		if ( ! $this->options->has_credentials( $api_credentials ) ) {
```

Comment:
`Smaily_For_WP_Options::has_credentials` does not have arguments any more.

Reply:
db1825d